sdjournal/journal.go

```diff
@@ -291,6 +296,13 @@ func (j *Journal) Wait(timeout time.Duration) int {
 	return int(r)
 }
 
+// WaitIndefinitely will synchronously wait until the journal gets changed.  This
+// call will sleep indefinitely while it waits for a change.
+func (j *Journal) WaitIndefinitely() int {
+	indef, _ := time.ParseDuration(maxDuration)
+	return j.Wait(indef)
+}
```

Comment:
The Wait() implementation needs to change to detect when the sentinel value was supplied and supply the 0xffff... value to sd_journal_wait() when it does.  I didn't mean for us to just supply a large value to Wait() and treat that as unlimited, there is likely an implementation difference underneath - one registers a timer, the other doesn't.  Let's not involve a timer when we're interested in waiting forever.

Please document the sentinel in the Wait() comment as well.

As for WaitIndefinitely(), I don't think we need it, just pass the public sentinel value to Wait() for the indefinite wait.